src/gpu/intel/ref_matmul.cl

```diff
@@ -81,7 +109,11 @@ __kernel void ref_matmul(__global SRC_DATA_T *A, __global WEI_DATA_T *B,
     int mb = get_global_id(2);
 
 #if WITH_DST_ZPOINTS
+#if HOST_DST_ZP
```

Comment:
I think we can limit the changes to the interface and kernel prologue, for example:

```c++
    // Point pointers to the scalar arguments.
#if HOST_SRC_ZP
    const SRC_ZP_DATA_T *a0 = &a0_val;
#endif
#if HOST_DST_ZP
    const int *c0 = &c0_val;
#endif
    ...

    // Kernel body is the same.
    int dst_zp = c0[0];
```

This should help to keep the kernel itself less macro-heavy.

Reply:
Yeah, others have that long offset.

But I would still follow the approach as I suggested. These are reference kernels so compactness/readability is important while performance is less important vs optimized kernels. So avoiding extra macros in the kernel body would be nice.